runtimes/node-18.0/server.js

```diff
@@ -1,85 +1,226 @@
-const path = require("path");
+const fs = require("fs");
 const micro = require("micro");
-const { json, send } = require("micro");
+const { text, json, send } = require("micro");
 
 const USER_CODE_PATH = '/usr/code-start';
 
 const server = micro(async (req, res) => {
-    const body = await json(req);
-
-    if (req.headers[`x-internal-challenge`] !== process.env['INTERNAL_RUNTIME_KEY']) {
-        return send(res, 500, {stderr: 'Unauthorized'});
+    if (req.headers[`x-open-runtimes-secret`] !== process.env['INTERNAL_RUNTIME_KEY']) {
+        return send(res, 500, 'Unauthorized');
     }
 
-    const request = {
-        variables: body.variables ?? {},
-        headers: body.headers ?? {},
-        payload: body.payload ?? '',
+    const logs = [];
+    const errors = [];
+    const response = {
+        headers: {},
+        body: '',
+        statusCode: 200,
+
+        cookies: {},
+        contentType: null,
     };
 
-    console.stdlog = console.log.bind(console);
-    console.stderror = console.error.bind(console);
-    console.stdinfo = console.info.bind(console);
-    console.stddebug = console.debug.bind(console);
-    console.stdwarn = console.warn.bind(console);
+    const contentType = req.headers['content-type'] ?? 'text/plain';
+    const body = contentType === 'application/json' ? await json(req) : await text(req);
```

Comment:
What about other content types? What if a user wants to handle a form submission with an Appwrite function? In this case the content type would be either `multipart/form-data` or `application/x-www-form-urlencoded`

Reply:
Added support for form-data and url-encoded body.